applications/StructuralMechanicsApplication/custom_constitutive/user_provided_linear_elastic_law.cpp

```diff
@@ -0,0 +1,285 @@
+        // Get Values to compute the constitutive law:
+        Flags& r_flags = rParameterValues.GetOptions();
+
+        // Previous flags saved
+        const bool flag_stress = r_flags.Is( ConstitutiveLaw::COMPUTE_STRESS );
+
+        // We compute the stress
+        r_flags.Set( ConstitutiveLaw::COMPUTE_STRESS, true );
+        UserProvidedLinearElasticLaw<TDim>::CalculateMaterialResponseCauchy(rParameterValues);
+        rValue = rParameterValues.GetStressVector();
+
+        // Previous flags restored
+        r_flags.Set( ConstitutiveLaw::COMPUTE_STRESS, flag_stress );
+    }
+
+    return( rValue );
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<unsigned int TDim>
+Matrix& UserProvidedLinearElasticLaw<TDim>::CalculateValue(
+    ConstitutiveLaw::Parameters& rParameterValues,
+    const Variable<Matrix>& rThisVariable,
+    Matrix& rValue
+    )
+{
+    if (rThisVariable == CONSTITUTIVE_MATRIX || rThisVariable == CONSTITUTIVE_MATRIX_PK2 || rThisVariable == CONSTITUTIVE_MATRIX_KIRCHHOFF) {
+        this->CalculateElasticMatrix(rValue, rParameterValues);
+    }
+
+    return( rValue );
+}
+
+//*************************CONSTITUTIVE LAW GENERAL FEATURES *************************
+/***********************************************************************************/
+
+template<unsigned int TDim>
+void UserProvidedLinearElasticLaw<TDim>::GetLawFeatures(Features& rFeatures)
+{
+    //Set the type of law
+    rFeatures.mOptions.Set(INFINITESIMAL_STRAINS);
+
+    //Set strain measure required by the consitutive law
+    rFeatures.mStrainMeasures.push_back(StrainMeasure_Infinitesimal);
+    rFeatures.mStrainMeasures.push_back(StrainMeasure_Deformation_Gradient);
+
+    //Set the strain size
+    rFeatures.mStrainSize = GetStrainSize();
+
+    //Set the spacedimension
+    rFeatures.mSpaceDimension = WorkingSpaceDimension();
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<unsigned int TDim>
+int UserProvidedLinearElasticLaw<TDim>::Check(
+    const Properties& rMaterialProperties,
+    const GeometryType& rElementGeometry,
+    const ProcessInfo& rCurrentProcessInfo
+    )
+{
+    KRATOS_CHECK_VARIABLE_KEY(ELASTICITY_TENSOR);
+    return 0;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<unsigned int TDim>
+void UserProvidedLinearElasticLaw<TDim>::CalculateElasticMatrix(
+    Matrix& rConstitutiveMatrix,
+    ConstitutiveLaw::Parameters& rValues
+    )
+{
+    const Properties& r_material_properties = rValues.GetMaterialProperties();
+    rConstitutiveMatrix = r_material_properties[ELASTICITY_TENSOR];
```

Comment:
```suggestion
    noalias(rConstitutiveMatrix) = r_material_properties[ELASTICITY_TENSOR];
```
I forgot this one too